python/vcastDataServer.py

```diff
@@ -25,27 +25,27 @@
 
 def init_application(logFilePath):
     app = Flask(__name__)
-
     with app.app_context():
 
-        @app.route("/ping")
+        @app.route("/ping", methods=["POST"])
         def pingRoute():
             return ping()
 
-        @app.route("/shutdown")
+        @app.route("/shutdown", methods=["POST"])
         def shutdownRoute():
             return shutdown()
 
-        @app.route("/runcommand")
+        @app.route("/runcommand", methods=["POST"])
         def runcommandRoute():
-            clientRequestText = request.args.get("request")
+            # Data from the request is a stringyfied json
+            clientRequestText = request.data.decode("utf-8")
             clientRequest = decodeRequest(clientRequestText)
+            # Ensure clientRequest is correctly decoded or processed
             return runcommand(clientRequest, clientRequestText)
 
-        # Note: this string must match what is in vcastAdapter.ts -> startServer()
```

Comment:
Restore comment

Reply:
Restored comment in c9c845a8b485173cc75c581d64e8f10b58eeb68e.